datalad/support/gitrepo.py

```diff
@@ -772,6 +817,14 @@ def _from_line(cls, line):
         )
 
 
+InfoT = TypeVar("InfoT", FetchInfo, PushInfo)
+
+
+class GitAddOutput(TypedDict):
+    file: str
+    success: bool
```

Comment:
note to myself and other @datalad/developers  -- `TypedDict` is what would save our sanity in formalizing our return record dictionaries, which we produce using https://github.com/datalad/datalad/blob/HEAD/datalad/interface/results.py#L51 .  `TypedDict` allows for `NotRequired` (not sure why not `Optional`) annotation for keys which do not have to be present.